samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/assessment/AssessmentEntityProducer.java

```diff
@@ -413,16 +413,26 @@ public Map<String, String> transferCopyEntities(String fromContext, String toCon
 		try {
 			if (cleanup) {
 				log.debug("deleting assessments from {}", toContext);
+				// Delete all draft assessments
 				AssessmentService service = new AssessmentService();
 				List<AssessmentData> assessmentList = service.getAllActiveAssessmentsbyAgent(toContext);
-                log.debug("found {} assessments in site: {}", assessmentList.size(), toContext);
+                log.debug("found {} draft assessments in site: {}", assessmentList.size(), toContext);
                 for (AssessmentData oneassessment : assessmentList) {
-                    log.debug("removing assessemnt id = {}", oneassessment.getAssessmentId());
+                    log.debug("removing draft assessment id = {}", oneassessment.getAssessmentId());
                     service.removeAssessment(oneassessment.getAssessmentId().toString());
                 }
+                
+                // Delete all published assessments
+                PublishedAssessmentService publishedAssessmentService = new PublishedAssessmentService();
+                List<PublishedAssessmentData> publishedAssessmentList = publishedAssessmentService.getAllPublishedAssessmentsForSite(toContext);
+                log.debug("found {} published assessments in site: {}", publishedAssessmentList.size(), toContext);
+                for (PublishedAssessmentData publishedAssessment : publishedAssessmentList) {
+                    log.debug("removing published assessment id = {}", publishedAssessment.getPublishedAssessmentId());
+                    publishedAssessmentService.removeAssessment(publishedAssessment.getPublishedAssessmentId().toString());
+                }
```

Comment:
so is this putting the published assessments in the trash where they can be recovered?

Reply:
yes